sdk/anomalydetector/azure-ai-anomalydetector/src/samples/java/com/azure/ai/anomalydetector/MultivariateSample.java

```diff
@@ -85,105 +91,103 @@ private static AnomalyDetectorClient getClient(String endpoint, String key) {
+        return model.getModelInfo();
     }
 
-    private static UUID getResultId(AnomalyDetectorClient client, BinaryData body, UUID modelId) {
-        RequestOptions requestOptions = new RequestOptions();
-        Response<BinaryData> response = client.detectMultivariateBatchAnomalyWithResponse(modelId.toString(), body, requestOptions);
-        String responseBodyStr = response.getValue().toString();
-        JsonObject jsonObject = Json.createReader(new StringReader(responseBodyStr)).readObject();
-        UUID resultId = UUID.fromString(jsonObject.getString("resultId"));
+    private static UUID getResultId(AnomalyDetectorClient client, DetectionRequest body, UUID modelId) {
+        DetectionResult detectionResult = client.detectMultivariateBatchAnomaly(modelId.toString(), body);
+        UUID resultId = UUID.fromString(detectionResult.getResultId());
         return resultId;
     }
 
-    private static String getInferenceStatus(AnomalyDetectorClient client, UUID resultId) {
-        RequestOptions requestOptions = new RequestOptions();
-        Response<BinaryData> response = client.getMultivariateBatchDetectionResultWithResponse(resultId.toString(), requestOptions);
-        String responseBodyStr = response.getValue().toString();
-        JsonObject jsonObject = Json.createReader(new StringReader(responseBodyStr)).readObject();
-        String status = jsonObject.getJsonObject("summary").getString("status");
-        return status;
+    private static DetectionStatus getInferenceStatus(AnomalyDetectorClient client, UUID resultId) {
+        DetectionResult detectionResult = client.getMultivariateBatchDetectionResult(resultId.toString());
+        return detectionResult.getSummary().getStatus();
     }
 
     private static void getModelList(AnomalyDetectorClient client, Integer skip, Integer top) {
-        RequestOptions requestOptions = new RequestOptions()
-            .addQueryParam("skip", skip.toString())
-            .addQueryParam("top", top.toString());
-        PagedIterable<BinaryData> response = client.listMultivariateModels(requestOptions);
-
-        Iterator<PagedResponse<BinaryData>> ite = response.iterableByPage().iterator();
+        PagedIterable<Model> response = client.listMultivariateModels(skip, top);
+        Iterator<PagedResponse<Model>> ite = response.iterableByPage().iterator();
         int i = 1;
         while (ite.hasNext()) {
-            PagedResponse<BinaryData> items = ite.next();
+            PagedResponse<Model> items = ite.next();
             System.out.println("The result in the page " + i);
             i++;
-            for (BinaryData item : items.getValue()) {
-                JsonObject jsonObject = Json.createReader(new StringReader(item.toString())).readObject();
-                System.out.println("\t" + jsonObject.getString("modelId"));
+            for (Model item : items.getValue()) {
+                System.out.println("\t" + item.getModelId());
             }
         }
```

Comment:
See if you can do this via stream API. https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html

Reply:
updated.